### PR TITLE
Adds spinner for chapter group progress

### LINKED ISSUE #570: On Item > progress > chapter view: wrong transient message
This "chapter view for user" message has nothing to do here. 
Check why it is here and fix it (for instance, display a loader if it is waiting for a response).

Screenshot 2021-07-28 at 16 55 21

### PR DESCRIPTION
**Pull request by Iloveall**
Fixes #570

### CODE REVIEW

File: src/app/modules/item/pages/chapter-group-progress/chapter-group-progress.component.html [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
+    [itemData]="itemData"
+    [group]="state.data"
+  >
+  </alg-group-progress-grid>
+
+  <div class="spinner" *ngIf="state.isFetching">
+    <p-progressSpinner [style]="{ width: '3.3333rem', height: '3.3333rem' }"></p-progressSpinner>
+  </div>
+
+  <ng-template #userChapterView>
```
Comment (smadbe) [inline src/app/modules/item/pages/chapter-group-progress/chapter-group-progress.component.html:13]: `#userChapterView` no used anymore?

File: src/app/modules/item/pages/chapter-group-progress/chapter-group-progress.component.html [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
+    *ngIf="state.isReady"
+    [itemData]="itemData"
+    [group]="state.data"
+  >
+  </alg-group-progress-grid>
+
+  <div class="spinner" *ngIf="state.isFetching">
+    <p-progressSpinner [style]="{ width: '3.3333rem', height: '3.3333rem' }"></p-progressSpinner>
+  </div>
+
```
Comment (smadbe) [inline src/app/modules/item/pages/chapter-group-progress/chapter-group-progress.component.html:12]: and what about the error case? Not directly related to the issue, I know, but related with the use of the state and important.

**Review by smadbe — CHANGES_REQUESTED** [review]

File: src/app/modules/item/pages/chapter-group-progress/chapter-group-progress.component.html (hunk 1 above)
Comment (Iloveall) [reply src/app/modules/item/pages/chapter-group-progress/chapter-group-progress.component.html:13]: Fixed

File: src/app/modules/item/pages/chapter-group-progress/chapter-group-progress.component.html (hunk 2 above)
Comment (Iloveall) [reply src/app/modules/item/pages/chapter-group-progress/chapter-group-progress.component.html:12]: Fixed

**Review by smadbe — APPROVED** [review]

**Pull request merged**